src/platform/iotjs_module_gpio-linux-general.inl.h

```diff
@@ -78,139 +62,123 @@ bool SetPinDirection(int32_t pin, GpioDirection direction) {
     strcpy(buffer, "out");
   }
 
-  DDDLOG("GPIO SetPinDirection() - path: %s, dir: %s", direction_path, buffer);
+  DDDLOG("%s - path: %s, dir: %s", __FUNCTION__, direction_path, buffer);
 
-  return DeviceOpenWriteClose(direction_path, buffer);
+  return iotjs_systemio_open_write_close(direction_path, buffer);
 }
 
 
 // FIXME: Implement SetPinMode()
-bool SetPinMode(int32_t pin, GpioMode mode) {
-  IOTJS_ASSERT(mode != kGpioModeNone && mode != kGpioModeOpendrain);
-
+static bool gpio_set_mode(int32_t pin, GpioMode mode) {
   return true;
 }
 
 
-#define GPIO_WORKER_INIT_TEMPLATE(initialized)                                \
-  IOTJS_ASSERT(iotjs_gpio_initialized() == initialized);                      \
-  iotjs_gpio_reqwrap_t* req_wrap = iotjs_gpio_reqwrap_from_request(work_req); \
-  iotjs_gpio_reqdata_t* req_data = iotjs_gpio_reqwrap_data(req_wrap);
+bool iotjs_gpio_write(int32_t pin, bool value) {
+  char value_path[GPIO_PATH_BUFFER_SIZE];
+  snprintf(value_path, GPIO_PATH_BUFFER_SIZE, GPIO_PIN_FORMAT_VALUE, pin);
 
+  char buffer[2] = { 0 };
+  buffer[0] = value ? '1' : '0';
 
-void InitializeGpioWorker(uv_work_t* work_req) {
-  GPIO_WORKER_INIT_TEMPLATE(false);
-  DDDLOG("GPIO InitializeGpioWorker()");
+  DDDLOG("%s - pin: %d, value: %d", __FUNCTION__, pin, value);
 
-  // Check if GPIO interface exits.
-  if (DeviceCheckPath(GPIO_INTERFACE)) {
-    req_data->result = kGpioErrOk;
-  } else {
-    req_data->result = kGpioErrInitialize;
-  }
+  return iotjs_systemio_open_write_close(value_path, buffer);
 }
 
 
-void ReleaseGpioWorker(uv_work_t* work_req) {
-  GPIO_WORKER_INIT_TEMPLATE(true);
-  DDDLOG("GPIO ReleaseGpioWorker()");
+int iotjs_gpio_read(int32_t pin) {
+  char buffer[GPIO_VALUE_BUFFER_SIZE];
+  char value_path[GPIO_PATH_BUFFER_SIZE];
+  snprintf(value_path, GPIO_PATH_BUFFER_SIZE, GPIO_PIN_FORMAT_VALUE, pin);
 
-  req_data->result = kGpioErrOk;
+  if (!iotjs_systemio_open_read_close(value_path, buffer,
+                                      GPIO_VALUE_BUFFER_SIZE - 1)) {
+    return -1;
+  }
 
-  const iotjs_environment_t* env = iotjs_environment_get();
-  uv_loop_t* loop = iotjs_environment_loop(env);
+  return atoi(buffer);
```

Comment:
If the read value is NULL, it returns an error. Therefore, `atoi` is not executed.